### ISSUE TITLE
PV-Opt stuck at "Optimising Charging Plan"

### ISSUE THREAD

**Issue by punkymuzzle** [stevebuk1/pv_opt#309]
**Describe the bug**
PV-Opt stuck at "Optimising Charging Plan".  It was working fine until this evening but it seems to have crashed.
So charging slots are not being allocated as pv-opt is stuck

**To Reproduce**
Not sure how to reproduce - I get this error when I reboot HA, if I amend a pv-opt file causing it to restart, or if I start AppDaemon.

**Expected behaviour**
Should work as designed.

**Screenshots**
![image](https://github.com/user-attachments/assets/2eb1004d-aa04-48f8-bced-337c1655b5e7)

AppDaemon log seems OK:
![image](https://github.com/user-attachments/assets/76c23950-4526-4d36-a2b7-125ac785712a)

**Desktop (please complete the following information):**
-RasPi 4
![image](https://github.com/user-attachments/assets/e36c078c-762d-4a6f-9164-f658f50c2863)

**Smartphone (please complete the following information):**
N/A

**Additional context**
Was working fine until earlier today (late afternoon  / early evening on 25/11/24

Logs attached:
[error.log](https://github.com/user-attachments/files/17909947/error.log)
[main.log](https://github.com/user-attachments/files/17909948/main.log)
[pv_opt.log](https://github.com/user-attachments/files/17909949/pv_opt.log)

History of the pv-opt status shows it stopped working approx 20:30 this evening:
![image](https://github.com/user-attachments/assets/3f93bc21-5c96-43ba-8d8b-e9e2abfb07da)

**Comment by stevebuk1** [comment]
You are the 2nd user to report a hangup last night just past the point where the Solcast data is loaded, which is where it loads prices from the Octopus website. I suspect its something to do with that. Has it recovered now?

**Comment by punkymuzzle** [comment]
Just got back in from working away so only just able to reply now.
Yes it now seems to be working.
After I reported the issue I installed the previous version of pv-opt, but it still crashed.  So I reinstalled the latest version with the same issue.
left it overnight and checked this morning but still broken; so I left it and went to work.
When I checked later in the day it had started working again, so all is good now.
Thanks for replying to this, appreciate it.
I'll close this issue.
Thanks again
